extend/generic-writer/configuration/configuration.md

```diff
@@ -0,0 +1,718 @@
+
+#### Headers
+
+[OPTIONAL] Allows you to define default query parameters that are being sent with each request. 
+
+**NOTE** That you can reference parameters defined in `user_parameters` using the `{"attr":"SOME_KEY"}` syntax.
+
+```json
+"request_options": {
+"api_request": {
+                "method": "POST",
+                "endpoint_path": "/customer",
+                "headers": {
+                              "Last-updated": 123343534
+                           },
+                ...
+```
+See [example 006](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/006-simple-json-custom-headers/)
+
+#### Query parameters
+
+[OPTIONAL] Allows you to define default query parameters that are being sent with each request.
+
+**NOTE** That you can reference parameters defined in `user_parameters` using the `{"attr":"SOME_KEY"}` syntax.
+
+```json
+        "request_options": {
+            "api_request": {
+                "method": "POST",
+                "endpoint_path": "/customer/[[id]]",
+                "query_parameters": {
+                    "dryRun":true,
+                    "date": {
+                        "attr": "date"
+                    }
+                }
+            }
+```
+
+See [example 009](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/009-simple-json-request-parameters/)
+
+
+### Content
+
+[REQUIRED] Defines how to process the input and how the sent content should look like.
+
+#### Content Type
+
+[REQUIRED]  Defines how the input table is translated to a request:
+
+- `JSON` - input table is converted into a JSON (see `json_mapping`) sent as `application/json` type. See [example 001](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/001-simple-json/)
+- `JSON_URL_ENCODED` - input table is converted into a JSON and sent as `application/x-www-form-urlencoded`. See [example 021](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/021-simple-json-url-encoded-form/)
+- `BINARY` - input table is sent as binary data (just like `curl --data-binary`). See [example](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/tests/functional/binary_simple/)
+- `BINARY-GZ` - input is sent as gzipped binary data. See [example](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/tests/functional/binary_gz/)
+- `EMPTY_REQUEST` - sends just empty requests. Usefull for triggerring webhooks, DELETE calls, etc. 
+As many requests as there are rows on the input are sent. Useful with `iterate_by_columns` enabled to trigger multiple endpoints. 
+See [example 022](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/022-empty-request-iterations-delete/) 
+
+
+```json
+        "request_options": {
+            ...,
+            "content": {
+                "content_type": "JSON",
+            ....
+```
+
+#### JSON Mapping
```

Comment:
here something is missing - JSON mapping applies to the JSON type, what options apply to the other types?